### DISCUSSION TITLE
Unify cloud SDK using Kotlin

### DISCUSSION THREAD

**Discussion by hfhbd** [SAP/cloud-sdk-js#4167] [Ideas]
Hey,
I am just curios, you do offer a JS and a Java based SDK. 
Did you ever think about using Kotlin multiplatform to write the SDKs only once? This would reduce the current effort to implement a new api or fix bugs, because you would have only one code base. Of course the migration would take some time, but Kotlin/JS has plain JS/typescript interop as well as Kotlin/JVM has Java interop.

Greetings from WOB
Philip

**Comment by marikaner** [comment]
Hey @hfhbd,

thank you for your suggestion!
Yes, we have thought and talked about it. However, currently there are few behaviors that significantly differ due to the different nature of Java/JS (e.g. request on the thread context in Java vs. request as a parameter in JS). We also believe that there are still quite a lot of paradigms that differ between Java and JavaScript - a common solution for both could potentially suffer from worse user experience on at least one side.
That said, we still would not be completely opposed to at least look deeper into this to see how well the issues above could be mitigated, however at this point in time - on a product this far along, we think the time and effort we would have to put into this would probably not pay off.
